### ISSUE TITLE
Is `width` a parameter of the Python object `vim.current.window`?

### ISSUE THREAD

**Issue by alphaCTzo7G** [vim/vim#2997]
I am trying to use PYthon to interface with VIM in my vimrc. I am aware that
this can be done in `vimrc` in VIML, but I don't want to learn VIML too much,
and am already familiar with `Python`.

I looked into some of the examples here: http://vimhelp.appspot.com/if_pyth.txt.html#python-window

Its mentioned here the `width` is a parameter of any `window` object.

However, when I do `py3 print(dir(vim.current.window))`, it gives me the
following output:
`['__dir__', 'buffer', 'col', 'cursor', 'height', 'number', 'options', 'row',
'tabpage', 'v alid', 'vars']`.

This doesn't contain `width` as one of the `attributes` of `current.window`,
however, when I use the following `ex` command:
`:py3 print(vim.current.window.width)`

It does give me a output of `90`, so it seems `width` is indeed an attribute of
`vim.current.window`.

Is this a bug or how this is supposed to happen?

**Comment by alphaCTzo7G** [comment]
I did search through the repository to find where the interface might be located, but my initial search just resulted in bunch of `.vim` files.

**Comment by k-takata** [comment]
See `src/if_python3.c` and `src/if_py_both.h` to know how the python 3 interface is implemented.

**Comment by k-takata** [comment]
Hmm, it looks there is a mistake in the implementation:

```diff
--- a/src/if_py_both.h
+++ b/src/if_py_both.h
@@ -3838,7 +3838,7 @@ get_firstwin(TabPageObject *tabObject)
 }
 static char *WindowAttrs[] = {
     "buffer", "cursor", "height", "vars", "options", "number", "row", "col",
-    "tabpage", "valid",
+    "tabpage", "valid", "width",
     NULL
 };
 
```

**Comment by alphaCTzo7G** [comment]
Ok.. thanks for letting me know. I can make the `patch`.. but probably you know more about what steps need to be followed. Let me know if you need help..

**Comment by brammool** [comment]
Ken Takata wrote:

> Hmm, it looks there is a mistake in the implementation:
> 
> ```diff
> --- a/src/if_py_both.h
> +++ b/src/if_py_both.h
> @@ -3838,7 +3838,7 @@ get_firstwin(TabPageObject *tabObject)
>  }
>  static char *WindowAttrs[] = {
>      "buffer", "cursor", "height", "vars", "options", "number", "row", "col",
> -    "tabpage", "valid",
> +    "tabpage", "valid", "width",
>      NULL
>  };
>  
> ```

Thanks.  Let's order the items as they are used in the WindowAttr()
function, that makes it a bit easier to keep in sync.

-- 
Tips for aliens in New York: Land anywhere.  Central Park, anywhere.
No one will care or indeed even notice.
		-- Douglas Adams, "The Hitchhiker's Guide to the Galaxy"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by k-takata** [comment]
Fixed with 8.1.0041: https://github.com/vim/vim/commit/e950f9992b291a07e4fb41cb561156f5b382cd5d
